lib/connections/service.go

```diff
@@ -36,6 +36,8 @@ var (
 	listeners = make(map[string]listenerFactory, 0)
 )
 
+const perDeviceWarningRate = 1.0 / (15 * 60) // Once per 15 minutes
```

Comment:
At least for the GUI, we stash the last five warnings and they're quite infrequent. So it should be available directly in the GUI when opened if it's happened at all recently